server/SCHEMAS/mongodb.js

```diff
@@ -38,104 +43,270 @@ const QandASchema = new Schema({
-  let answers = question[0].answers
-
-  // setup the new answer data
-  let answer = new Answer({
-    answer_id: data.id,
-    answer_body: data.body,
-    answer_date: data.date_written,
-    answerer_name: data.answerer_name,
-    answerer_email: data.answerer_email,
-    answer_reported: data.reported,
-    answer_helpfulness: data.helpful,
-    answer_photos: []
-  })
-
-  // if not present, insert the answer to the array from the question
-  let isPresent = false
-  answers.forEach(answer => {
-    if (!isPresent) {
-      if (answer.answer_id === parseInt(data.id)) {
-        isPresent = true
+  try {
+    let query = { question_id }
+    let question = await QandA.find(query).lean()
+    let id = question[0]._id
+    let answers = question[0].answers
+
+    let answer = new Answer({
+      answer_id: data.id,
+      answer_body: data.body,
+      answer_date: data.date_written,
+      answerer_name: data.answerer_name,
+      answerer_email: data.answerer_email,
+      answer_reported: data.reported,
+      answer_helpfulness: data.helpful,
+      answer_photos: data.photos || []
+    })
+
+    let isPresent = false
+    answers.forEach(answer => {
+      if (!isPresent) {
+        if (answer.answer_id === parseInt(data.id)) {
+          isPresent = true
+        }
       }
-    }
-  })
+    })
+
+    if (!isPresent) {
+      answers.push(answer)
 
-  if (!isPresent) {
-    answers.push(answer)
-    // add the answer to the asnwers array for this question
-    await QandA.findByIdAndUpdate(id, { answers })
+      await QandA.findByIdAndUpdate(id, { answers }).lean()
+    }
   }
+  catch (err) { return err }
 }
 
+// add an image to a given answer
 let answerPhotoInsert = async (data) => {
-  let url = data.url
-
-  // query the db for the answer and needed info
-  let question = await QandA.findOne({
-    'answers': {
-      $elemMatch: {
-        'answer_id': data.answer_id
+  try {
+    let url = data.url
+    let question = await QandA.findOne({
+      'answers': {
+        $elemMatch: {
+          'answer_id': data.answer_id
+        }
       }
-    }
-  })
+    }).lean()
 
-  let id = question._id
-  let answers = question.answers
-  let newAnswer, photosArr
+    let id = question._id
+    let answers = question.answers
+    let newAnswer, photosArr
 
-  answers.forEach((answer, index) => {
-    if (parseInt(data.answer_id) === answer.answer_id) {
-      newAnswer = answer
-      photosArr = answer.answer_photos
-      answers.splice(index, 1)
-    }
-  })
+    answers.forEach((answer, index) => {
+      if (parseInt(data.answer_id) === answer.answer_id) {
+        newAnswer = answer
+        photosArr = answer.answer_photos
+        answers.splice(index, 1)
+      }
+    })
 
-  // if not present, insert the answer to the array from the question
-  let isPresent = false
-  photosArr.forEach(photoUrl => {
-    if (!isPresent) {
-      if (photoUrl === url) {
-        isPresent = true
+    let isPresent = false
```

Comment:
probably $addToSet could be used here too

Reply:
Thanks for this! I implemented tonight and saw some minute improvement on performance while testing :)